packages/headless/src/controllers/folded-result-list/headless-folded-result-list.ts

```diff
@@ -160,6 +172,16 @@ export function buildFoldedResultList(
       );
     },
 
+    findResultByUniqueID(id) {
+      return this.state.results.find((r) => r.result.uniqueId === id);
+    },
+
+    findResultByCollection(collection) {
+      return this.state.results.find(
+        (r) => r.result.raw.foldingcollection === collection
+      );
+    },
```

Comment:
@olamothe suggested we add these, the idea was to pass a result instead of the strings initially but I thought it made more sense to pass the id/collection.
what do you think, Olivier?

Reply:
Passing in the id or collection as a string does not seem really useful

As an end user of the lib, with no extensive knowledge of Coveo, how would you know which string property to use ? There's many of them available on each individual result.

You would have to read the doc, which is not that awesome. It puts burden on the user.

Where it starts to show value is if it accept a generic object (the whole result), and then internally know which property of that result it should use out of those available. 

That way, using TypeScript, you also get re-assurance that you are doing "the right thing", VS a string type that is non specific, so you are never sure which property you should pass in.

It should also recursively search through all descendants, as Benjamin said.

With all that, it starts being pretty useful.

Reply:
ok I see, that all makes sense!